DataCollatorIntegrationTest::test_nsp failing on GPU

@sgugger I believe you are the right person to tag?

```python
=================================== FAILURES ===================================
_____________________ DataCollatorIntegrationTest.test_nsp _____________________
[gw0] linux -- Python 3.7.6 /home/hf/actions-runner_transformers/_work/transformers/transformers/.env/bin/python

self = <tests.test_data_collator.DataCollatorIntegrationTest testMethod=test_nsp>

    @slow
    def test_nsp(self):
        tokenizer = AutoTokenizer.from_pretrained("bert-base-cased")
        data_collator = DataCollatorForNextSentencePrediction(tokenizer)
    
        dataset = TextDatasetForNextSentencePrediction(tokenizer, file_path=PATH_SAMPLE_TEXT, block_size=512)
        examples = [dataset[i] for i in range(len(dataset))]
        batch = data_collator(examples)
        self.assertIsInstance(batch, dict)
    
        # Since there are randomly generated false samples, the total number of samples is not fixed.
        total_samples = batch["input_ids"].shape[0]
        self.assertEqual(batch["input_ids"].shape, torch.Size((total_samples, 512)))
        self.assertEqual(batch["token_type_ids"].shape, torch.Size((total_samples, 512)))
>       self.assertEqual(batch["masked_lm_labels"].shape, torch.Size((total_samples, 512)))
E       KeyError: 'masked_lm_labels'
```

https://github.com/huggingface/transformers/runs/1303479917?check_suite_focus=true

--- a/tests/test_data_collator.py
+++ b/tests/test_data_collator.py
@@ -175,7 +175,7 @@ class DataCollatorIntegrationTest(unittest.TestCase):
         total_samples = batch["input_ids"].shape[0]
         self.assertEqual(batch["input_ids"].shape, torch.Size((total_samples, 512)))
         self.assertEqual(batch["token_type_ids"].shape, torch.Size((total_samples, 512)))
-        self.assertEqual(batch["masked_lm_labels"].shape, torch.Size((total_samples, 512)))
+        self.assertEqual(batch["labels"].shape, torch.Size((total_samples, 512)))
         self.assertEqual(batch["next_sentence_label"].shape, torch.Size((total_samples,)))
 
     @slow

